add method to fetch users collection items by release

AnssiAhola opened the pull request:
WIP

Implements [this API point](https://www.discogs.com/developers/#page:user-collection,header:user-collection-collection-items-by-release) as requested by @raratiru in #90
closes #90 

@JOJ0 @alifhughes If either of you could tell me what you think

Usage
```python

release_instances = me.collection_items(22155985)
for instance in release_instances:
    print(instance) 
```
Once you have the instance you can get the collection folder it belong to and delete it (or do whatever)
```python
folder = me.collection_folders[instance.folder_id]
folder.remove_release(instance)
```
Not the most elegant but thats how the discogs api is sometimes :woman_shrugging:

JOJ0 commented:
Hi, playling around with it just now. Great find @raratiru, didn't know this API exists. Thanks a lot for the draft @AnssiAhola, it's gonna be a useful addition to the library IMHO!

I think the placing of the method in the User model is good. Right next to collection_folders, ok.
You have another idea where it could possibly fit @AnssiAhola ?

The name of the method is long but actually making clear what it's about. I don't have any better ideas except that spelling could be equally to other collection_.. methods in this place: `collection_items_by_release`

Or just some very similar names like: `collection_instances_by_release()`, `collection_items_of_release()` `collection_instances_of()`, `collection_items_of`,

AnssiAhola commented:
> I think the placing of the method in the User model is good. Right next to collection_folders, ok. You have another idea where it could possibly fit @AnssiAhola ?

Now that I think about it I guess it fits quite well in there :smile:
 
> The name of the method is long but actually making clear what it's about. I don't have any better ideas except that spelling could be equally to other collection_.. methods in this place: `collection_items_by_release`
> 
> Or just some very similar names like: `collection_instances_by_release()`, `collection_items_of_release()` `collection_instances_of()`, `collection_items_of`,

How about
```python
collection_items_by_release(release)
```
OR
```python
collection_items_by(release)
```
OR
```python
collection_items_of(release)
```

Do you have any suggestions/preference @raratiru ?

JOJ0 commented:
And now the most radical approach: Why not simply `collection_items(release_id)`? What it is exactly would be documented in the docstring (-> autogenerated module docs) and in our enduser docs anyway. Somewhere here:

https://python3-discogs-client.readthedocs.io/en/latest/discogs_client.html#discogs_client.models.User.collection_folders
https://python3-discogs-client.readthedocs.io/en/latest/fetching_data.html#collection-data

AnssiAhola commented:
Updated the method name to 
```python
collection_items
```
Also added comment to explain what the method does.

Any examples or better description should written in the docs i think.

JOJ0 commented:
If you follow the syntax of numpy docstring style for args and return values it will render correctly in sphinx autodoc. Sorry on mobile now but youll find nice examples of this style in the Client model class...I think...

JOJ0 commented:
@AnssiAhola sorry that was wrong and misleading. I ment the Fetchers class not the Client class. Have a look here, this is how the docstring needs to be formatted to properly render with Sphinx autodoc: https://github.com/joalla/discogs_client/blob/master/discogs_client/fetchers.py#L21-L50

JOJ0 commented:
Perfect! Let's merge this!

AnssiAhola commented:
I'll commit those last changes later today and then I'll merge
